pkg/model/api/v1/ephemeraldashboard_test.go

```diff
@@ -0,0 +1,556 @@
+                "matchers": [
+                  "up"
+                ]
+              }
+            }
+          }
+        }
+      ],
+      "panels": {
+        "MyPanel": {
+          "kind": "Panel",
+          "spec": {
+            "display": {
+              "name": "simple line chart"
+            },
+            "plugin": {
+              "kind": "TimeSeriesChart",
+              "spec": {
+                "showLegend": false,
+                "lines": [
+                  "up"
+                ]
+              }
+            }
+          }
+        }
+      },
+      "layouts": [
+        {
+          "kind": "Grid",
+          "spec": {
+            "items": [
+              {
+                "x": 0,
+                "y": 0,
+                "width": 3,
+                "height": 4,
+                "content": {
+                  "$ref": "#/spec/panels/MyPanel"
+                }
+              }
+            ]
+          }
+        }
+      ],
+      "duration": "6h",
+      "refreshInterval": "30s"
+    }
+  }
+}`
+	panel := &Panel{
+		Kind: "Panel",
+		Spec: PanelSpec{
+			Display: PanelDisplay{
+				Name: "simple line chart",
+			},
+			Plugin: common.Plugin{
+				Kind: "TimeSeriesChart",
+				Spec: map[string]interface{}{
+					"lines": []interface{}{
+						"up",
+					},
+					"showLegend": false,
+				},
+			},
+		},
+	}
+	expected := &EphemeralDashboard{
+		Kind: KindEphemeralDashboard,
+		Metadata: ProjectMetadata{
+			Metadata: Metadata{
+				Name: "SimpleDashboard",
+			},
+			Project: "perses",
+		},
+		Spec: EphemeralDashboardSpec{
+			TTL: model.Duration(24 * time.Hour),
+			DashboardSpec: DashboardSpec{
+				Variables: []dashboard.Variable{
+					{
+						Kind: variable.KindList,
+						Spec: &dashboard.ListVariableSpec{
+							ListSpec: variable.ListSpec{
+								Plugin: common.Plugin{
+									Kind: "PrometheusLabelNamesVariable",
+									Spec: map[string]interface{}{
+										"matchers": []interface{}{
+											"up",
+										},
+									},
+								},
+							},
+							Name: "labelName",
+						},
+					},
+					{
+						Kind: variable.KindList,
+						Spec: &dashboard.ListVariableSpec{
+							ListSpec: variable.ListSpec{
+								Plugin: common.Plugin{
+									Kind: "PrometheusLabelValuesVariable",
+									Spec: map[string]interface{}{
+										"labelName": "$labelName",
+										"matchers": []interface{}{
+											"up",
+										},
+									},
+								},
+							},
+							Name: "labelValue",
+						},
+					},
+				},
+				Panels: map[string]*Panel{"MyPanel": panel},
+				Layouts: []dashboard.Layout{
+					{
+						Kind: dashboard.KindGridLayout,
+						Spec: &dashboard.GridLayoutSpec{
+							Items: []dashboard.GridItem{
+								{
+									X:      0,
+									Y:      0,
+									Width:  3,
+									Height: 4,
+									Content: &common.JSONRef{
+										Ref:    "#/spec/panels/MyPanel",
+										Path:   []string{"spec", "panels", "MyPanel"},
+										Object: panel,
```

Comment:
Object is actually equal to nil after the Unmarshal process of the DashboardSpec

Reply:
I don't get it, this part is basically copy-pasted from the same test on Dashboard side, and it does reference the panel object:
![image](https://github.com/perses/perses/assets/7058693/423c830c-4258-4c43-ac70-7b4d41585a46)

Reply:
that's because the code checking and creating the reference is executed at the Dashboard struct level not at the spec level.

This is because if you execute at the spec level, then you don't know the exact json path.